src/fieldAnalyser/typeUtils.spec.ts

```diff
@@ -0,0 +1,47 @@
+import {FieldTypes} from '@coveord/platform-client';
+import {getGuessedTypeFromValue, isValidTypeTransition} from './typeUtils';
+
+describe('typeUtils', () => {
+  const signedIntegerLimit = 2147483647;
```

Comment:
If you want to be very explicit:
```suggestion
  const signedIntegerLimit = 0b1111111111111111111111111111111;
```